internal/sirius/create_warning_test.go

```diff
@@ -0,0 +1,153 @@
+package sirius
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/pact-foundation/pact-go/dsl"
+	"github.com/stretchr/testify/assert"
+)
+
+func newPact() *dsl.Pact {
+	return &dsl.Pact{
+		Consumer:          "sirius-lpa-frontend",
+		Provider:          "sirius",
+		Host:              "localhost",
+		PactFileWriteMode: "merge",
+		LogDir:            "../../logs",
+		PactDir:           "../../pacts",
+	}
+}
+
+func TestCreateWarning(t *testing.T) {
+	pact := newPact()
+	defer pact.Teardown()
+
+	testCases := []struct {
+		name          string
+		setup         func()
+		cookies       []*http.Cookie
+		expectedError func(int) error
+	}{
+		{
+			name: "OK",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("A donor exists").
+					UponReceiving("A request to create a warning").
+					WithRequest(dsl.Request{
+						Method: http.MethodPost,
+						Path:   dsl.String("/api/v1/persons/189/warnings"),
+						Body: dsl.Like(map[string]interface{}{
+							"warningType": "Complaint Received",
+							"warningText": "Some warning notes",
+						}),
+						Headers: dsl.MapMatcher{
+							"X-XSRF-TOKEN":        dsl.String("abcde"),
+							"Cookie":              dsl.String("XSRF-TOKEN=abcde; Other=other"),
+							"OPG-Bypass-Membrane": dsl.String("1"),
+							"Content-Type":        dsl.String("application/json"),
+						},
+					}).
+					WillRespondWith(dsl.Response{
+						Status:  http.StatusCreated,
+						Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+					})
+			},
+			cookies: []*http.Cookie{
+				{Name: "XSRF-TOKEN", Value: "abcde"},
+				{Name: "Other", Value: "other"},
+			},
+		},
+		{
+			name: "Validation error",
```

Comment:
We haven't been doing pacts for errors, should we?

Reply:
Not sure. There's definitely value in checking the shape of the error response (to make sure we can read from `validation_errors`), but I feel a bit uncomfortable contracting specific validation scenarios, because now we're effectively checking business logic inside a consumer.

On a related note, I'd quite like to move the auth layer into the provider test suite so that we don't have to specify the cookies inside the Pact contracts. If we ever change the auth, it'll be a real PITA updating all these tests.

Reply:
That makes sense, I'll remove them so it is like our others

Reply:
The auth stuff shouldn't cause a problem as it is all handled in pact-proxy, so we'd be able to handle all the changes there. Whether continuing to do it this way gives us any value 🤷

Reply:
Yeah, I think handling it in pact-proxy would be better and would keep the contract focussed on data. We'd probably need to supply a header like `Sirius-Role: System Admin` to indicate who to auth as, which might also help with those tests where we want to mock different responses for different roles.